### DISCUSSION TITLE
Polymorphic Types

### DISCUSSION THREAD

**Discussion by t1** [eclipse/microprofile-open-api#658] [General]
I'm not sure if I missed something, but [OpenAPI](https://spec.openapis.org/oas/v3.1.0.html#models-with-polymorphism-support) as well as [JSON-B](https://jakarta.ee/specifications/jsonb/3.0/jakarta-jsonb-spec-3.0#polymorphic-types) and [Jackson](https://github.com/FasterXML/jackson-databind/wiki/Databind-Annotations#polymorphic-type-handling) support polymorphic types, but I didn't find anything in MP Open-API.

I know that this often over-used, but there are actual cases, where it's very helpful.

Did I miss something?

**Comment by Azquelt** [comment]
The MP OpenAPI spec isn't at all specific about how a Jakarta REST application should be turned into an OpenAPI document, and so it's up to implementations as to exactly what they want to implement.

Unless your implementation has better support for this, you may need to use the `@Schema` and `@DiscriminatorMapping` annotations to describe your polymorphic type, in addition to whatever JSON-B or Jackson annotations are required.

I would like the spec to be much more prescriptive about how classes and annotations are converted to OpenAPI schema, but it's a big task with lots of edge cases and I don't have time to work on it.
